### PR TITLE
Mem alloc UT

### PR DESCRIPTION
**Pull request by JeromeGalan**
This PR adds 2 features :
-  Unit_test tools functions.
-  All unit tests for memory allocator.

Github Action scripts will failed because LuosHal x86 feature is not merged to master.
First, reviewers have to :
- Review LuosHAL "unit_test" branch (a PR is already opened).
- Merge this LuosHAL branch to master when everything's OK.
- Then, Github Action scripts should be correct for Luos unit tests...

### CODE REVIEW

**Review by Salem-Luos — COMMENTED** [review]
put a max of little schematics at the beginning of test that's the best way to understand the code

can you create a little doc on notion with all the schematic on tx msg to check if we don't forget a case

File: test/unittest_transport.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#ifndef UNITTEST_TRANSPORT_H
```
[the remark this answers is not in the document]
Comment (JeromeGalan) [reply test/unittest_transport.h:1]: Yes it is, in order to launch platformIO UNITY test engine. If this file is removed, UNITY crashes.

File: test/memory_allocator_static/unit_test_mem_alloc_static.c [hunk 2]
```diff
@@ -0,0 +1,952 @@
...
+#include "../test/unit_test.h"
+#include "../Robus/inc/robus_struct.h"
+#include "../src/msg_alloc.c"
+
+void unittest_DoWeHaveSpace(void)
+{
+    NEW_TEST_CASE("\t* There is enough space");
+    MsgAlloc_Init(NULL);
+    {
+        NEW_STEP();
```
[the remark this answers is not in the document]
Comment (JeromeGalan) [reply test/memory_allocator_static/unit_test_mem_alloc_static.c]: "New step" function is explained in Notion Document (coming soon on Notion)...

**Comment by JeromeGalan** [comment]
> put a max of little schematics at the beginning of test that's the best way to understand the code
> 
> can you create a little doc on notion with all the schematic on tx msg to check if we don't forget a case

- Schematics have been added.
- Doc will be deliver soon on Notion.

**Review by nicolas-rabault — APPROVED** [review]
Good to me!
Can't wait to be victimized by your tests :)

**Review by nicolas-rabault — COMMENTED** [review]
Why there is a binary file test/CodeCoverage/Readme ?
It seems to aprears OK on the real folder. Github interprets it as binary but it's a good text file.

**Comment by JeromeGalan** [comment]
> Why there is a binary file test/CodeCoverage/Readme ?
> It seems to aprears OK on the real folder. Github interprets it as binary but it's a good text file.

I'm not sure to understand your remark.
You mean this "Readme" file is good but there is a problem in Github interpretation ?  Github interprets this text file as binary but it shouldn't ?

**Comment by nicolas-rabault** [comment]
Yes, but anyway everything seems good...

**Pull request closed without merging**